packages/react-core/src/components/Pagination/Pagination.tsx

```diff
@@ -201,8 +202,9 @@ export class Pagination extends React.Component<PaginationProps, { ouiaStateId:
   };
 
   getLastPage() {
-    const { itemCount, perPage } = this.props;
-    return Math.ceil(itemCount / perPage) || 0;
+    const { itemCount, perPage, page } = this.props;
+    // when itemCount is not known let's set lastPage as page+1 as we don't know the total count
+    return itemCount ? Math.ceil(itemCount / perPage) || 0 : page + 1;
```

Comment:
`itemCount` needs to be updated to `(itemCount || itemCount === 0)` for the case of 0 items, rather than unknown items.